cosim/black-parrot-example/ps.cpp

```diff
@@ -29,6 +29,23 @@
 #define BP_NCPUS 1
 #endif
 
+// Offsets for GP0 (Check bsg_zynq_pl_shell in top_zynq)
+#define GP0_CSR_RESET          0x0
+#define GP0_CSR_DRAM_INITED    0x4
+#define GP0_CSR_DRAM_BASE      0x8
+#define GP0_PL_TO_PS_FIFO_DATA 0xC
+#define GP0_PL_TO_PS_FIFO_CTRS 0x10
+#define GP0_PS_TO_PL_FIFO_CTRS 0x14
+#define GP0_MINSTRET           0x18 // 64-bit
+#define GP0_MEM_PROFILER_0     0x20
+#define GP0_MEM_PROFILER_1     0x24
+#define GP0_MEM_PROFILER_2     0x28
+#define GP0_MEM_PROFILER_3     0x2C
+
+// BP Address in PS
+#define BP_L2_ADDR  0x80000000U
```

Comment:
So there are actually 3 addresses here

#define DRAM_BASE_ADDR 0x80000000
#define DRAM_MAX_ALLOC_SIZE 0x20000000
#define GP1_DRAM_BASE_ADDR GP1_ADDR_BASE
#define GP1_CSR_BASE_ADDR GP1_ADDR_BASE + DRAM_MAX_ALLOC_SIZE

The difference is that for Z2, GP1_ADDR_BASE = 0x80000000, but for u96v2 GP1_ADDR_BASE = 0x500000000. BP_DRAM_BASE_ADDR is used for NBF code so 

```
      if (nbf[1] >= DRAM_BASE_ADDR)
        base_addr = GP1_ADDR_BASE - DRAM_BASE_ADDR;
      else
        base_addr = GP1_ADDR_BASE + DRAM_MAX_ALLOC_SIZE;
```

but then the rest of the code should reference GP1_DRAM_BASE_ADDR or GP1_CSR_BASE_ADDR + the appropriate offsets. Does that make sense?

For clarity, this will be compatible with the changes here: https://github.com/black-parrot-hdk/zynq-parrot/pull/43/files for both Pynq Z2 and Ultra96v2 boards (and other boards that we add